forgevtt-module.js

```diff
@@ -2326,7 +2326,19 @@ class ForgeVTT_FilePicker extends FilePicker {
         if (target == null) {
             target = "";
             if (body.uuid) {
-                target = `worlds/${game.world.id || game.world.name}`
+                target = `worlds/${game.world.id || game.world.name}`;
+            }
+            if (event && event.type === "paste") {
+                // For a paste event from an unnamed clipboard image, the browser will use a generic name.
+                // It is not possible to get original filename/info from the clipboard data.
+                // There is no clear convention for name that allows us to tell when this happens.
+                // "image.png" is most common, but use a broad regex that might catch other OS/browsers behaviours.
+                const regex = /^(image|img|clipboard|screen(?:shot|[\s_-]shot)).*\.(png|jpg|jpeg|gif|svg|webp)$/i;
+                if (file && file.name && regex.test(file.name)) {
```

Comment:
Why the regexp? If it's a paste event, then it shouldn't matter what the filename is, consider it unnamed and always use the timestamp, otherwise it could overwrite thing if there's a paste that doesn't match the regexp.
I can't think of a situation where you're pasting an image somewhere and expect it to overwrite some other file.

Reply:
> Why the regexp? If it's a paste event, then it shouldn't matter what the filename is, consider it unnamed and always use the timestamp, otherwise it could overwrite thing if there's a paste that doesn't match the regexp. I can't think of a situation where you're pasting an image somewhere and expect it to overwrite some other file.

Copying a file from your local hard drive is not an unnamed paste and has a specific name. In this case, it would upload to that specific asset, find that it already exists, and not create a duplicate.

Would you prefer it always create a unique, even if the same image is copy and pasted?

Reply:
> Copying or dragging a named file from your local hard drive is not an unnamed paste and has a specific name.

But would that have an event of type "paste" ? I'm not sure, and I trust you know more than me on this specific issue. I just raised concerns that appeared to me, but I'll let you be the judge of what's the best approach.

Reply:
> > Copying or dragging a named file from your local hard drive is not an unnamed paste and has a specific name.
> 
> But would that have an event of type "paste" ? I'm not sure, and I trust you know more than me on this specific issue. I just raised concerns that appeared to me, but I'll let you be the judge of what's the best approach.

Aha, I see what you mean!
Yeah, a drag event would indeed have event.type `drag` rather than `paste`. `drag` events always have a known name.
Copy-pasting from your local drive will have event.type `paste`, and will also have a known name.

That said, I like what [9de9d22](https://github.com/ForgeVTT/fvtt-module-forge-vtt/pull/32/commits/9de9d2239ce3db58f678e2c82d5470d233348f62) does since it's the least complex, doesn't rely on `event`, and is the least deviation from core Foundry, which always creates a new asset for the event regardless of `paste` or `drag`.